### PR TITLE
Yeet the empty outcomes on exit of builder

### PR DESCRIPTION
**Pull request by mfranz1**
This PR fulfills story: https://app.shortcut.com/clarkcan/story/7899/have-builder-not-save-empty-outcomes-on-exit
An outcome will be deleted on the exit of the learning object builder if that outcome has no text.

image
image

### CODE REVIEW

File: src/app/onion/learning-object-builder/components/outcome/outcome-typeahead/outcome-typeahead.component.ts [hunk 1]
```diff
@@ -100,9 +95,11 @@ export class OutcomeTypeaheadComponent implements OnInit, OnChanges, OnDestroy {
...
         }
 
-        if (this.goodVerb) {
-          this.enteredText.emit(this.text);
-        }
+        // we want to always emit changes from the outcome fields even if they are null
+        // there's a function that will validate each field and will throw an error if needed
+        this.enteredText.emit(this.text);
+        this.selectedVerb.emit(this.verb);
+        this.selectedCategory.emit(this.bloom);
```
Comment (EVogel1999) [inline src/app/onion/learning-object-builder/components/outcome/outcome-typeahead/outcome-typeahead.component.ts:102]: Does this work when you input an incorrect verb such as 'asdf'?

**Review by EVogel1999 — COMMENTED** [review]
Just have a few questions but it looks good!

File: src/app/onion/learning-object-builder/components/outcome/outcome.component.ts [hunk 2]
```diff
@@ -93,21 +91,15 @@ export class OutcomeComponent implements OnInit {
...
-    }
+    this.selectedLevel.emit(val);
   }
 
   emitText(val) {
-    if (val) {
-      this.textChanged.emit(val);
-    }
+    this.textChanged.emit(val);
   }
```
[the remark this answers is not in the document]
Comment (mfranz1) [reply src/app/onion/learning-object-builder/components/outcome/outcome.component.ts:103]: Yes but there is logic in place that will prevent undefined values being saved to the outcome for a verb, bloom, and text.

File: src/app/onion/learning-object-builder/components/outcome/outcome-typeahead/outcome-typeahead.component.ts (hunk 1 above)
Comment (mfranz1) [reply src/app/onion/learning-object-builder/components/outcome/outcome-typeahead/outcome-typeahead.component.ts:102]: Yes, we emit the verb, but the builder-store handles the validation logic

**Review by EVogel1999 — APPROVED** [review]

File: src/app/onion/learning-object-builder/builder-store.service.ts [hunk 3]
```diff
@@ -825,6 +825,23 @@ export class BuilderStore {
       });
   }
 
+  /**
+   * Public function to remove empty outcomes from a learning object
+   */
+  public async removeEmptyOutcomes() {
+    // Get most up-to-date values for current learning object
+    await this.fetch(this.learningObject.id);
```
Comment (StarChaser44) [inline src/app/onion/learning-object-builder/builder-store.service.ts:833]: Where does the learning object id that is fetched go when this request is completed?

**Review by Cwagne17 — APPROVED** [review]

**Review by JosueMolinaMorales — APPROVED** [review]

**Review by jouypo23 — APPROVED** [review]

**Pull request merged**